libs/hdf-converters/src/xccdf-results-mapper.ts

```diff
@@ -392,16 +450,34 @@ export class XCCDFResultsMapper extends BaseConverter {
               {
                 data: {
                   path: ['check.check-content-ref.name'],
-                  transformer: parseHtml
+                  transformer: (text: string | string[]) =>
+                    asArray(text).join('\n') || undefined
                 },
                 label: 'check'
               },
               {
                 data: {
-                  path: ['fixtext', 'fix'],
-                  transformer: parseHtml
+                  path: ['fixtext.text', 'fixtext', 'fix.text', 'fix'],
+                  transformer: (text: string | string[]) =>
+                    asArray(text).map(parseHtml).join('\n') || undefined
```

Comment:
the '|| undefined' is dangerous cause it makes it not match the schema

xccdf-scc-rhel8-hdf.json

![image](https://github.com/mitre/heimdall2/assets/2015821/ffcc8c78-91be-43d8-bb72-eb1e2e4a025f)

schema: https://github.com/mitre/heimdall2/blob/master/libs/inspecjs/src/generated_parsers/v_1_0/exec-json.ts#L224

if a description is provided it needs to have both a label and data.  the or defined makes it so that sometimes only a label is provided.

Reply:
Updated! :+1: